base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
+
+# temp definitions while Transpose is not subtype of AbstractMatrix
+# Base.eltype{T}(A::Transpose{T}) = T
+
+# end temp
+
+Base.size(A::Transpose) = size(A.data, 2), size(A.data, 1)
+function Base.size(A::Transpose, i::Integer)
+    if i <= 0
+        error("arraysize: dimension out of range")
+    elseif i <= 2
+        return size(A.data, 3 - i)
+    else
+        return 1
+    end
+end
+
+@inline function Base.getindex(A::Transpose, i::Integer, j::Integer)
+    aji = A.data[j,i]
+    return ifelse(A.conjugated, conj(aji), aji)
+end
+
+Base.similar(A::Transpose, T::Type, d::Dims) = similar(A.data, T, d)
+
+ctranspose(A::AbstractMatrix) = Transpose{eltype(A), typeof(A)}(A, true)
+function ctranspose(A::Transpose)
+    if A.conjugated
+        return A.data
+    else
+        return map(conj, A.data)
+    end
+end
+transpose(A::AbstractMatrix) = Transpose{eltype(A), typeof(A)}(A, false)
+function transpose(A::Transpose)
+    if A.conjugated
+        return map(conj, A.data)
+    else
+        return A
+    end
+end
+
+transpose(x::AbstractVector) = [ transpose(v) for i=1, v in x ]
+ctranspose{T}(x::AbstractVector{T}) = T[ ctranspose(v) for i=1, v in x ] #Fixme comprehension
```

Comment:
will be lovely to have a `CovariantVector` type someday, so there can be consistency in view vs copy. But I agree this may be too much of a change now.